core/src/main/java/com/google/zxing/qrcode/encoder/MatrixUtil.java

```diff
@@ -271,12 +271,7 @@ static void embedDataBits(BitArray dataBits, int maskPattern, ByteMatrix matrix)
   // - findMSBSet(1) => 1
   // - findMSBSet(255) => 8
   static int findMSBSet(int value) {
-    int numDigits = 0;
-    while (value != 0) {
-      value >>>= 1;
-      ++numDigits;
-    }
-    return numDigits;
+    return 32 - Integer.numberOfLeadingZeros(value);
```

Comment:
Use highestOneBit? The only concerns are not using anything that is not present in Java 7 and Android's 4.0.x API but the two you use appear fine.